contracts/MetadataViews.cdc

```diff
@@ -737,4 +737,202 @@ pub contract MetadataViews {
         return Traits(traits)
     }
 
+    /// A struct to expose license information according to https://forum.onflow.org/t/flow-nft-license-project/4716
+    /// By default, there are only 12 possible license combinations,
+    /// so these are enabled by the 12 utility methods below the struct definition
+    /// to allow easy construction of each possible combination
+    /// For a given NFT, returning a license view is as simple as
+    /// calling the method for the desired view, like so:
+    ///
+    /*   pub fun resolveView(_ view: Type): AnyStruct? {
+            switch view {
+                case Type<MetadataViews.NFTLicense>():
+                    return MetadataViews.nlpVoteCom()
+                case ...
+    */
+    ///
+    /// The first version of the NFT license standard is only deployed
+    /// with the license names and text descriptions. Projects must refer
+    /// to the proposal and license documentation for specific descriptions of
+    /// the rights that each license grants.
+    ///
+    pub struct NFTLicense {
+        /// Array of the specific license identifiers
+        pub let licenses: [String]
+
+        /// Link to a badge image that can be displayed alongside
+        /// the asset on a webpage for a visual representation of the license
+        pub var badgeLink: Media?
+
+        /// Link to a description of the specific rights that the license offers
+        pub var descriptionLink: Media?
+
+        /// Plain text description of the license
+        pub var description: String?
+
+        init() {
+            self.licenses = []
+            self.badgeLink = nil
+            self.descriptionLink = nil
+            self.description = nil
+        }
+
+        access(contract) fun setBadgeLink(link: Media) {
+            self.badgeLink = link
+        }
+
+        access(contract) fun setDescriptionLink(link: Media) {
+            self.descriptionLink = link
+        }
+
+        access(contract) fun setDescription(_ description: String) {
+            self.description = description
+        }
+
+        access(contract) fun personalUse(): NFTLicense {
+            self.licenses.append("NLP-PER")
+            return self
+        }
+
+        access(contract) fun votingRights(): NFTLicense {
+            self.licenses.append("NFL-VOTE")
+            return self
+        }
+
+        access(contract) fun commercialRights(): NFTLicense {
+            self.licenses.append("NFL-COM")
+            return self
+        }
+
+        access(contract) fun additionalContentExperienceRights(): NFTLicense {
+            self.licenses.append("NLP-ALP")
+            return self
+        }
+
+        access(contract) fun merchandisingRights(): NFTLicense {
+            self.licenses.append("NLP-MERCH")
+            return self
+        }
+
+        access(all) fun equals(_ otherLicense: NFTLicense): Bool {
```

Comment:
This works. I didn't realize until I just tried it, but since `licenses` is an array of `String`, we can compare them directly - e.g. `self.licenses == otherLicense.licenses`

Reply:
oh, nice. I can change that